src/MissionManager/QGCMapPolygon.cc

```diff
@@ -311,6 +311,7 @@ void QGCMapPolygon::removeVertex(int vertexIndex)
 
     QObject* coordObj = _polygonModel.removeAt(vertexIndex);
     coordObj->deleteLater();
+    selectVertex(-1);
```

Comment:
Why clear a selected vertex if that vertex has not been removed?

Reply:
We do not know beforehand if vertex index > selected index or equal or below so simple functionality is just deselect all. But You are right, can make it more comprehensive to keep selected.

Reply:
Fixed: https://github.com/mavlink/qgroundcontrol/pull/8813/commits/a842c1a4f7e1a3c00504abfe6d267f061758e521